src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java

```diff
@@ -28,15 +47,70 @@ public WarehouseDto findById(Long warehouseId) {
                 .orElseThrow(() -> new ResourceNotFoundException("Warehouse with id=" + warehouseId + " doesn't exist"));
     }
 
-    // TODO: 11/4/20
+    @Transactional
     @Override
-    public WarehouseDto save(WarehouseDto dto) {
-        return null;
+    public WarehouseDto save(Long customerId, WarehouseDto warehouseDto) {
+        if (warehouseDto.getIdentifier() == null) {
+            throw new ResponseStatusException(
+                    HttpStatus.NOT_ACCEPTABLE, "\"Identifier\" can't be null");
+        }
```

Comment:
We can put @Column(nullable = false) in entity class to avoid setting null field and getting exception message.
In another hand we usually use Objects.nonNull().